sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/GenericArrayData.scala

```diff
@@ -71,6 +71,10 @@ class GenericArrayData(val array: Array[Any]) extends ArrayData {
   override def getArray(ordinal: Int): ArrayData = getAs(ordinal)
   override def getMap(ordinal: Int): MapData = getAs(ordinal)
 
+  override def setNullAt(ordinal: Int): Unit = throw new UnsupportedOperationException();
```

Comment:
it's pretty easy to implement it: `array(ordinal) = null`, why throw exception here?